packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -753,4 +762,423 @@ describe("AgreementService Integration Test", async () => {
+      attribute2 = { id: generateId() };
+      attribute3 = { id: generateId() };
+      attribute4 = { id: generateId() };
+      agreement1 = {
+        ...buildAgreement(eservice1.id, tenant1.id, agreementState.draft),
+        descriptorId: eservice1.descriptors[0].id,
+        producerId: eservice1.producerId,
+        certifiedAttributes: [attribute1, attribute2],
+        declaredAttributes: [attribute3],
+      };
+
+      agreement2 = {
+        ...buildAgreement(eservice1.id, tenant2.id, agreementState.active),
+        descriptorId: eservice1.descriptors[1].id,
+        producerId: eservice1.producerId,
+        declaredAttributes: [attribute3],
+        verifiedAttributes: [attribute4],
+      };
+
+      agreement3 = {
+        ...buildAgreement(eservice2.id, tenant1.id, agreementState.pending),
+        descriptorId: eservice2.descriptors[0].id,
+        producerId: eservice2.producerId,
+      };
+
+      agreement4 = {
+        ...buildAgreement(
+          eservice2.id,
+          tenant2.id,
+          agreementState.missingCertifiedAttributes
+        ),
+        // upgradeable agreement based on descriptors, but not in an upgradeable state
+        descriptorId: eservice2.descriptors[1].id,
+        producerId: eservice2.producerId,
+      };
+
+      agreement5 = {
+        ...buildAgreement(eservice3.id, tenant1.id, agreementState.archived),
+        descriptorId: eservice3.descriptors[0].id,
+        producerId: eservice3.producerId,
+      };
+
+      agreement6 = {
+        ...buildAgreement(eservice3.id, tenant3.id, agreementState.rejected),
+        descriptorId: eservice3.descriptors[0].id,
+        producerId: eservice3.producerId,
+      };
+
+      await addOneAgreement(agreement1, postgresDB, agreements);
+      await addOneAgreement(agreement2, postgresDB, agreements);
+      await addOneAgreement(agreement3, postgresDB, agreements);
+      await addOneAgreement(agreement4, postgresDB, agreements);
+      await addOneAgreement(agreement5, postgresDB, agreements);
+      await addOneAgreement(agreement6, postgresDB, agreements);
+    });
+
+    it("should get all agreements if no filters are provided", async () => {
+      const allAgreements = await agreementService.getAgreements({}, 10, 0);
+      expect(allAgreements.totalCount).toEqual(6);
+      expect(allAgreements.results).toEqual(
+        expect.arrayContaining([
+          agreement1,
+          agreement2,
+          agreement3,
+          agreement4,
+          agreement5,
+          agreement6,
+        ])
+      );
+    });
+
+    it("should get agreements with filters: producerId", async () => {
+      const agreements1 = await agreementService.getAgreements(
+        {
+          producerId: eservice1.producerId,
+        },
+        10,
+        0
+      );
+      expect(agreements1.totalCount).toEqual(2);
+      expect(agreements1.results).toEqual(
+        expect.arrayContaining([agreement1, agreement2])
+      );
```

Comment:
```suggestion
      expect(agreements1).toEqual({
        totalCount: 2,
        results: expect.arrayContaining([agreement1, agreement2]),
      });
```

I would use the `toEqual` on the whole object in order to have a cleaner view on what we expect, same for the following cases